### PR TITLE
Relationship searching on tables with spaces

### PR DESCRIPTION
**Pull request by mike12345567**
## Description
It is possible to search for rows using the table name - this is something that data providers will often do - but if the external table name contained a space there could be an issue with the table name not lining up after it comes in through the API (in the body of the request).

Added a test case for this as well, only applies to external tables as the internal table name is not used for searching.

Addresses: https://github.com/Budibase/budibase/issues/15139

### CODE REVIEW

File: packages/backend-core/src/sql/utils.ts [hunk 1]
```diff
@@ -66,6 +66,14 @@ export function buildExternalTableId(datasourceId: string, tableName: string) {
   return `${datasourceId}${DOUBLE_SEPARATOR}${tableName}`
 }
 
+export function checkTableId(tableId: string) {
```
Comment (adrinr) [inline packages/backend-core/src/sql/utils.ts:69]: Should this not be encoding always? It will encode only if it is actually needed

**Review by samwho — APPROVED** [review]

File: packages/backend-core/src/sql/utils.ts (hunk 1 above)
Comment (adrinr) [inline packages/backend-core/src/sql/utils.ts:69]: Also, this function name is confusing. It does not check the table id, it encodes it

File: packages/server/src/api/controllers/row/utils/utils.ts [hunk 2]
```diff
@@ -65,19 +65,21 @@ export function getSourceId(ctx: Ctx): { tableId: string; viewId?: string } {
     const { sourceId } = ctx.params
     if (docIds.isViewId(sourceId)) {
       return {
-        tableId: utils.extractViewInfoFromID(sourceId).tableId,
+        tableId: sql.utils.checkTableId(
+          utils.extractViewInfoFromID(sourceId).tableId
```
Comment (adrinr) [inline packages/server/src/api/controllers/row/utils/utils.ts:69]: Should the `checkTableId` not be used internally on `extractViewInfoFromID`?

File: packages/backend-core/src/sql/utils.ts (hunk 1 above)
Comment (mike12345567) [reply packages/backend-core/src/sql/utils.ts:69]: true - I can adjust them both - I was just replicating how it builds the table IDs today!

**Review by adrinr — APPROVED** [review]

**Pull request merged**